src/controllers/CurrentAccountController.js

```diff
@@ -22,13 +24,20 @@ export class CurrentAccountController {
     this.store = new ObservableStore(
       Object.assign({}, defaults, options.initState)
     );
-    this.poller = undefined;
+
+    extension.alarms.onAlarm.addListener(({ name }) => {
+      if (name === 'updateBalances') {
+        this.updateBalances();
+      }
+    });
     this.restartPolling();
   }
 
   restartPolling() {
-    clearInterval(this.poller);
-    this.poller = setInterval(this.updateBalances.bind(this), POLL_INTERVAL);
+    extension.alarms.clear('updateBalances');
```

Comment:
There's no need to clear alarm before creation, because alarm with the same name will be replaced, according to docs:

https://developer.chrome.com/docs/extensions/reference/alarms/#method-create
https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/alarms/create

There are also other places where you do it.